### ISSUE TITLE
Network changed

### ISSUE THREAD

**Issue by richardschneider** [richardschneider/net-mdns#42]
@cerna got a `System.OutOfMemoryException` when running .Net Core 3.0 (preview) on a  Debian Stretch.  The [stack track](https://github.com/richardschneider/net-mdns/issues/22#issuecomment-446956519) indicates that `NetworkAddressChanged ` was involved.  While this may just be an issue with the preview release; it could mean something is wrong with our code.

I've never liked the [code](https://github.com/richardschneider/net-mdns/blob/master/src/MulticastService.cs#L272-L273) that removes and then adds the `NetworkAddressChanged` event handler.

**Comment by richardschneider** [comment]
@cerna I've made some fixes in the [network-changed branch](https://github.com/richardschneider/net-mdns/tree/network-changed).  Would you mind trying it out on your .Net Core 3.0 Debian host?

Also, did you try it against the latest .Net Core 2.2.101?

Thanks in advance.

**Comment by cerna** [comment]
@richardschneider I have tried it on my Debian host (completely the same as previous tests) and so far it's looking good running on .NET Core 3.0 Preview (3.0.100-preview-009812). I was not able to replicate the same exeption as yesterday, well, actually no exeption was thrown. (Tried with avahi running and announcing services to network, instance of Spike and instance of Browser running on the Debian host and multiple other systems running Spikes/Browsers and other Bonjour/Zeroconf programs in my local home network.)

And yes, I tried it against the 2.2.101 SDK (I have no other 2.2 version installed.) And against it even this branch is not working.
